dev/benchmark/add-index-with-load.md

```diff
@@ -0,0 +1,345 @@
+| 4     | 54 |  168.8 |           3320 |
+| 8     | 39 |  123.8 |           2475 |
+| 16    | 36 |   59.6 |           1213 |
+| 32    | 42 |   93.2 |           1835 |
+| 48    | 51 |   115.7 |           2261 |
+
+![add-index-load-1-b1024](../../media/add-index-load-1-b1024.png)
+
+#### `tidb_ddl_reorg_batch_size = 2048`
+
+| tidb_ddl_reorg_worker_cnt |  add_index_durations(s) | sysbench TPS   | sysbench QPS |
+| :------------------------ | :---------------------- | :------------- | :----------- |
+| 1     | 918 |  243.3 |           4855 |
+| 2     | 1160 |  209.9 |           4194 |
+| 4     | 342 |  185.4 |           3707 |
+| 8     | 1316 |  151.0 |           3027 |
+| 16    | 795 |   30.5 |           679 |
+| 32    | 1130 |   26.69 |           547 |
+| 48    | 893 |   27.5 |           552 |
+
+![add-index-load-1-b2048](../../media/add-index-load-1-b2048.png)
+
+#### `tidb_ddl_reorg_batch_size = 4096`
+
+| tidb_ddl_reorg_worker_cnt |  add_index_durations(s) | sysbench TPS   | sysbench QPS |
+| :------------------------ | :---------------------- | :------------- | :----------- |
+| 1     | 3042 |  200.0 |           4001 |
+| 2     | 3022 |  203.8 |           4076 |
+| 4     | 858 |  195.5 |            3971 |
+| 8     | 3015 |  177.1 |           3522 |
+| 16    | 837 |   143.8 |           2875 |
+| 32    | 942 |   114 |           2267 |
+| 48    | 187 |   54.2 |           1416 |
+
+![add-index-load-1-b4096](../../media/add-index-load-1-b4096.png)
+
+### 测试结论
+
+若 Add Index 的目标列正在进行较为频繁的写操作（本测试中涉及列的 update/insert/delete），默认 Add Index 配置对系统的线上负载有比较明显的影响，该影响主要来源于 Add Index 与 Column Update 并发进行造成的写冲突，系统的表现反应在：
+
+- 随着两个参数的逐渐增大，TiKV prewrite latch wait duration 有明显的升高，造成写入变慢。
+- `tidb_ddl_reorg_worker_cnt` 与 `tidb_ddl_reorg_batch_size` 非常大时，`admin show ddl` 命令可以看到 ddl job 的多次重试（例如 `Write conflict, txnStartTS 410327455965380624 is stale [try again later], ErrCount:38, SnapshotVersion:410327228136030220`），此时 Add Index 会持久非常久才能完成。
+
+## 测试方案2 -- Add Index 目标列不涉及写入（仅查询）
+
+1. 开始 `oltp_read_only` 测试。
+2. 与步骤 1 同时，使用 `alter table sbtest1 add index c_idx(c)` 添加索引。
+3. 在步骤 2 结束，即索引添加完成时，停止步骤 1。
+4. 获取指标：`alter table ... add index` 的运行时间；sysbench 在该时间段内的平均 TPS 和 QPS。
+5. 逐渐增大 `tidb_ddl_reorg_worker_cnt` 和 `tidb_ddl_reorg_batch_size` 两个参数，重复步骤 1-4。
+
+### 测试结果
+
+#### 无 add_index 时 oltp_read_only 结果
+
+| sysbench TPS | sysbench QPS    |
+| :------- | :-------- |
+| 550.9   | 8812.8 |
+
+#### `tidb_ddl_reorg_batch_size = 32`
+
+| tidb_ddl_reorg_worker_cnt |  add_index_durations(s) | sysbench TPS   | sysbench QPS |
+| :------------------------ | :---------------------- | :------------- | :----------- |
+| 1     | 376 |  548.9 |           8780 |
+| 2     | 212 |  541.5 |           8523 |
+| 4     | 135 |  538.6 |           8549 |
+| 8     | 114 |  536.7 |           8393 |
+| 16    | 77 |   533.9 |           8292 |
+| 32    | 46 |   533.4 |           8103 |
+| 48    | 46 |   532.2 |           8074 |
+
+![add-index-load-2-b32](../../media/add-index-load-2-b32.png)
```

Comment:
Please update the path for all images. This is to guarantee they can display well on the PingCAP website.

Reply:
Done, thanks